src/rdkafka_request.h

```diff
@@ -107,6 +107,41 @@ typedef struct rd_kafkap_NodeEndpoints_s {
         rd_kafkap_NodeEndpoint_t *NodeEndpoints;
 } rd_kafkap_NodeEndpoints_t;
 
+typedef struct rd_kafkap_produce_reply_tags_Partition_s {
+        int32_t Partition;
+        rd_kafkap_CurrentLeader_t CurrentLeader;
+} rd_kafkap_produce_reply_tags_Partition_t;
+
+typedef struct rd_kafkap_produce_reply_tags_Topic_s {
+        char *TopicName;
+        int32_t PartitionCnt;
+        rd_kafkap_produce_reply_tags_Partition_t *PartitionTags;
+} rd_kafkap_produce_reply_tags_Topic_t;
+
+typedef struct rd_kafkap_produce_reply_tags_s {
+        rd_kafkap_NodeEndpoints_t NodeEndpoints;
```

Comment:
When parsing it needs to keep track of the leader changes that are seen and only send the op if there was one change
```suggestion
        int32_t leader_change_cnt;
        rd_kafkap_NodeEndpoints_t NodeEndpoints;
```

Reply:
Updated

Reply:
This is not a Kafka protocol field but an additional field we need that's why librdkafka snake case

Reply:
Updated